### ISSUE TITLE
[Bug]: load collection into memory failed, caused by load segment failed, set segment to meta failed

### ISSUE THREAD

**Issue by zhoudayu123** [milvus-io/milvus#15831]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):cluster milvusdb/milvus-dev:master-20220302-f1aec49a
- SDK version(e.g. pymilvus v2.0.0rc2):java 2.0.4
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 5 * 4Core/16G Mem
- GPU: 
- Others:
```

### Current Behavior

install milvus cluster,
create collection,
insert data,
search data,
load collection success when first servel times , but ten minutes later, it failed always.

### Expected Behavior

load collection success, and search next.

### Steps To Reproduce

```markdown
code like:
1.create collection
HasCollectionParam param = HasCollectionParam.newBuilder()
                .withCollectionName("test_collection")
                .build();
        R<Boolean> booleanR = client.hasCollection(param);
        if (!booleanR.getData()){
            FieldType fieldType1 = FieldType.newBuilder()
                    .withName("thash")
                    .withDataType(DataType.BinaryVector)
                    .withDimension("192")
                    .build();
            FieldType fieldType2 = FieldType.newBuilder()
                    .withName("thash_id")
                    .withDataType(DataType.Int64)
                    .withPrimaryKey(true)
                    .withAutoID(true)
                    .build();
            CreateCollectionParam createCollectionReq = CreateCollectionParam.newBuilder()
                    .withCollectionName("test_collection")
                    .withDescription("test_collection")
                    .withShardsNum(2)
                    .addFieldType(fieldType1)
                    .addFieldType(fieldType2)
                    .build();
  
        client.createCollection(createCollectionReq);

2.insert data
                List<ByteBuffer> byteBuffers = new ArrayList<>();
                String hash = "a298f4505c2404d5ccbfc937a7455d2b62b3e1bab21188ad"

                byteBuffers.add(byteArray2ByteBuffer(hexStr2BinArr(hash)));
                List<InsertParam.Field> fields = new ArrayList<>();
   
                fields.add(new InsertParam.Field("thash", DataType.BinaryVector, byteBuffers));

                InsertParam insertParam = InsertParam.newBuilder()
                        .withCollectionName("test_collection")
                        .withFields(fields)
                        .build();
                client.insert(insertParam);
3. search 
           List<ByteBuffer> byteBufferList = new ArrayList<>();
           String hash  = "a298f4505c2404d5ccbfc937a7455d2b62b3e1bab21188ad";
            byteBufferList.add(byteArray2ByteBuffer(hexStr2BinArr(hash)));
           client.loadCollection(
                    LoadCollectionParam.newBuilder()
                            .withCollectionName("test_collection")
                            .build());
            SearchParam searchParam = SearchParam.newBuilder().withCollectionName("test_collection")
                    .withVectors(byteBufferList)
                    .withMetricType(MetricType.HAMMING)
                    .withVectorFieldName("thash")
                    .withTopK(20)
                    .withParams(new JsonObject().addProperty("nprobe", 32))
                    .withOutFields("thash_id")
                    .build();

            // execute search
            R<SearchResults> searchResponse = client.search(searchParam);
            log.info("searchResponse:{}", searchResponse.getData());
            client.releaseCollection(
                    ReleaseCollectionParam.newBuilder()
                            .withCollectionName("test_collection")
                            .build());

public static byte[] hexStr2BinArr(String hexString) {
        hexString = hexString.toLowerCase();
        int len = hexString.length() / 2;
        byte[] bytes = new byte[len];
        byte high = 0;
        byte low = 0;
        for (int i = 0; i < len; i++) {
            high = (byte) ((hexStr.indexOf(hexString.charAt(2 * i))) << 4);
            low = (byte) hexStr.indexOf(hexString.charAt(2 * i + 1));
            bytes[i] = (byte) (high | low);
        }
        return bytes;
    }

 public static ByteBuffer byteArray2ByteBuffer(byte[] value) {
        ByteBuffer buffer = ByteBuffer.wrap(value);
        buffer.position(value.length);
        return buffer;
    }
```

### Anything else?

![image](https://user-images.githubusercontent.com/51695477/156343561-2be442b5-5665-416c-b064-ba1a3e836cb2.png)
![image](https://user-images.githubusercontent.com/51695477/156343746-a49d780b-0ac5-4445-ada4-b68f63e7e057.png)

**Comment by yanliang567** [comment]
@soothing-rain could you please help on this issue?
/assign @soothing-rain 
/unassign

**Comment by zhoudayu123** [comment]
Hi, xiaofan. I found you have fixed this bug, and when will it be realease?

**Comment by yhmo** [comment]
> Hi, xiaofan @xiaofan-luan . I found you have fixed this bug, and when will it be realease?

The code change is still pending not merged, it will be available in the next minor version 2.0.2, which will be released in the middle of this month.

**Comment by xiaofan-luan** [comment]
> Hi, xiaofan @xiaofan-luan . I found you have fixed this bug, and when will it be realease?

Should be able to release on 2.0.2~
